### PR TITLE
Azure - Find VMs with a public IP address

### PR DESCRIPTION
**Pull request by tbarlow12**

### CODE REVIEW

File: docs/source/azure/examples/vmwithpublicip.rst [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+.. code-block:: yaml
+
+     policies:
+       - name: vms-with-public-ip
+         resource: azure.vm
+         filters:
+          - type: network-interface
+            key: 'properties.ipConfigurations[].properties.publicIPAddress.id'
+            op: eq
```
Comment (kapilt) [inline docs/source/azure/examples/vmwithpublicip.rst]: eq is the default, so a little redundant to document for an example. also not-null handling is a shortcut before the equality comparison.

Comment (stefangordon) [inline docs/source/azure/examples/vmwithpublicip.rst]: See not-null examples, can remove op entirely
http://capitalone.github.io/cloud-custodian/docs/filters.html

File: docs/source/azure/examples/vmwithpublicip.rst [hunk 2]
```diff
@@ -9,5 +9,3 @@ Find Virtual Machines with Public IP address
          filters:
           - type: network-interface
             key: 'properties.ipConfigurations[].properties.publicIPAddress.id'
-            op: eq
-            value: not-null
```
Comment (stefangordon) [inline docs/source/azure/examples/vmwithpublicip.rst]: Does this work? I thought you needed `value: not-null` and just no `op`

Comment (tbarlow12) [reply docs/source/azure/examples/vmwithpublicip.rst]: That would make more sense. I misunderstood yours and kapil's comments about that. I'll add it back in. Looks like the workers crashed anyway :)

**Comment by tbarlow12** [comment]
This resolves issue #2271

File: docs/source/azure/policy/resources/vm.rst [hunk 3]
```diff
@@ -20,6 +20,9 @@ Filters
   .. c7n-schema:: MetricFilter
        :module: c7n_azure.filters
 
+``network-interface``
+  Filter based on properties of the network interfaces associated with the virtual machine.
+
```
Comment (stefangordon) [inline docs/source/azure/policy/resources/vm.rst]: The auto-generated schema should be pulled in here like the authors

**Pull request merged**